Mathlib/Analysis/Calculus/MeanValue.lean

```diff
@@ -596,6 +596,32 @@ theorem eqOn_of_fderivWithin_eq (hs : Convex ℝ s) (hf : DifferentiableOn 𝕜
   refine hs.is_const_of_fderivWithin_eq_zero (hf.sub hg) (fun z hz => ?_) hx hy
   rw [fderivWithin_sub (hs' _ hz) (hf _ hz) (hg _ hz), sub_eq_zero, hf' _ hz]
 
+/-- If two functions have equal Fréchet derivatives at every point of a connected open set,
```

Comment:
It might also be useful to have the "integration is unique up to a constant" version of this, i.e. if two functions have equal derivatives, there exists some `C` with `f x = g x + C` for all `x`.